romana/cmd/root.go

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) 2016 Pani Networks
+// All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/romana/core/common"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+// Variables used for configuration and flags.
+var (
+	cfgFile  string
+	rootURL  string
+	version  bool
+	verbose  bool
+	format   string
+	platform string
+)
+
+// This represents the base command when called without any subcommands
+var RootCmd = &cobra.Command{
+	Use:   "romana",
+	Short: "Command line tools for romana services.",
+	Long: `Command line tools for romana services.
+
+Romana is a new Software Defined Network solution specifically
+designed for Cloud Native applications. Romana allows multi-tenant
+cloud computing networks for OpenStack, Docker and Kubernetes to
+be built without encapsulation or a virtual network overlay.
+
+Romana networks are less expensive to build, easier to operate
+and deliver higher performance than networks built using
+alternative overlay based SDN designs.
+
+For more information, please check http://romana.io
+`,
+}
+
+// Execute adds all child commands to the root command sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
```

Comment:
this is the entry point which function main() calls. here you add commands/subcommand mapping where control is passed around from main() to commands/subcommands evoked. updated the comment accordingly.